consensus/system_contract/consensus.go

```diff
@@ -0,0 +1,390 @@
+		return errInvalidExtra
+	}
+	//// All basic checks passed, verify cascading fields
+	return s.verifyCascadingFields(chain, header, parents)
+}
+
+// verifyCascadingFields verifies all the header fields that are not standalone,
+// rather depend on a batch of previous headers. The caller may optionally pass
+// in a batch of parents (ascending order) to avoid looking those up from the
+// database. This is useful for concurrently verifying a batch of new headers.
+func (s *SystemContract) verifyCascadingFields(chain consensus.ChainHeaderReader, header *types.Header, parents []*types.Header) error {
+	// The genesis block is the always valid dead-end
+	number := header.Number.Uint64()
+	if number == 0 {
+		return nil
+	}
+	// Ensure that the block's timestamp isn't too close to its parent
+	var parent *types.Header
+	if len(parents) > 0 {
+		parent = parents[len(parents)-1]
+	} else {
+		parent = chain.GetHeader(header.ParentHash, number-1)
+	}
+	if parent == nil || parent.Number.Uint64() != number-1 || parent.Hash() != header.ParentHash {
+		return consensus.ErrUnknownAncestor
+	}
+	if header.Time < parent.Time {
+		return errInvalidTimestamp
+	}
+	// Verify that the gasUsed is <= gasLimit
+	if header.GasUsed > header.GasLimit {
+		return fmt.Errorf("invalid gasUsed: have %d, gasLimit %d", header.GasUsed, header.GasLimit)
+	}
+	if !chain.Config().IsCurie(header.Number) {
+		// Verify BaseFee not present before EIP-1559 fork.
+		if header.BaseFee != nil {
+			return fmt.Errorf("invalid baseFee before fork: have %d, want <nil>", header.BaseFee)
+		}
+		if err := misc.VerifyGaslimit(parent.GasLimit, header.GasLimit); err != nil {
+			return err
+		}
+	} else if err := misc.VerifyEip1559Header(chain.Config(), parent, header); err != nil {
+		// Verify the header's EIP-1559 attributes.
+		return err
+	}
+
+	// only if block header comes from a new block msg, then verify the signature against the current signer
+	if header.IsNewBlock {
+		signer, err := ecrecover(header)
+		if err != nil {
+			return err
+		}
+
+		s.lock.RLock()
+		defer s.lock.RUnlock()
+
+		if signer != s.signerAddressL1 {
+			log.Error("Unauthorized signer", "Got", signer, "Expected:", s.signerAddressL1)
+			return ErrUnauthorizedSigner
+		}
+	}
+
+	return nil
+}
+
+// VerifyUncles implements consensus.Engine, always returning an error for any
+// uncles as this consensus mechanism doesn't permit uncles.
+func (s *SystemContract) VerifyUncles(chain consensus.ChainReader, block *types.Block) error {
+	if len(block.Uncles()) > 0 {
+		return errors.New("uncles not allowed")
+	}
+	return nil
+}
+
+// Prepare initializes the consensus fields of a block header according to the
+// rules of a particular engine. Update only timestamp and prepare ExtraData for Signature
+func (s *SystemContract) Prepare(chain consensus.ChainHeaderReader, header *types.Header) error {
+	header.BlockSignature = make([]byte, extraSeal)
+	header.IsEuclidV2 = true
+	header.Extra = []byte{}
```

Comment:
I think it's okay to set it to empty slice here, we should also remove it from the worker: 
https://github.com/scroll-tech/go-ethereum/blob/9139e949724c61d791a1a2942aeaddbbe74dcce0/eth/backend.go#L277

Reply:
Done.

Reply:
Would this not be an issue with Clique though?

Reply:
good point so we would need to make it conditional. let's just go back an leave it as it was. sorry! :D

Reply:
good point. I reverted it and added a comment.